integration/all/integration_test.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"testing"
+
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+	timeout                    = 10
+
+	K8sDistroAKSAzureLinux = "aks-AzureLinux"
+	K8sDistroAKSUbuntu     = "aks-Ubuntu"
+	K8sDistroARO           = "aro"
+	K8sDistroMinikubeGH    = "minikube-github"
+
+	securityProfileOperatorNamespace = "security-profiles-operator"
+	topTimeoutInSeconds              = 10
+)
+
+var (
+	// supportedContainerRuntimes = []string{ContainerRuntimeDocker, ContainerRuntimeContainerd, ContainerRuntimeCRIO}
+	supportedK8sDistros = []string{K8sDistroAKSAzureLinux, K8sDistroAKSUbuntu, K8sDistroARO, K8sDistroMinikubeGH}
+	cleaningUp          = uint32(0)
+
+	// containerRuntime = flag.String("container-runtime", "", "allows to do validation for expected runtime in the tests")
+	image          = flag.String("image", "", "gadget container image")
+	integration    = flag.Bool("integration", false, "run integration tests")
+	dnsTesterImage = flag.String("dnstester-image", "ghcr.io/inspektor-gadget/dnstester:latest", "dnstester container image")
+	testComponent  = flag.String("test-component", "", "run tests for specific component")
+
+	doNotDeployIG  = flag.Bool("no-deploy-ig", false, "don't deploy Inspektor Gadget")
+	doNotDeploySPO = flag.Bool("no-deploy-spo", false, "don't deploy the Security Profiles Operator (SPO)")
+
+	k8sDistro = flag.String("k8s-distro", "", "allows to skip tests that are not supported on a given Kubernetes distribution")
+	k8sArch   = flag.String("k8s-arch", "amd64", "allows to skip tests that are not supported on a given CPU architecture")
+
+	gadgetRepository = flag.String("gadget-repository", "ghcr.io/inspektor-gadget/gadget", "repository where gadget images are stored")
+	gadgetTag        = flag.String("gadget-tag", "latest", "tag used for gadgets's OCI images")
+)
```

Comment:
nit: Please separate this into three groups, flags specific for ig-k8s, inspektor-gadget and the ones in common.